diff --git a/L/LLVM/common.jl b/L/LLVM/common.jl
index 9162c2d14..df6e3e584 100644
--- a/L/LLVM/common.jl
+++ b/L/LLVM/common.jl
@@ -253,6 +253,8 @@ fi
 # Install things into $prefix, and make sure it knows we're cross-compiling
 CMAKE_FLAGS+=(-DCMAKE_INSTALL_PREFIX=${prefix})
 CMAKE_FLAGS+=(-DCMAKE_CROSSCOMPILING=True)
+# Prevent building native host tools, as we have already done that above
+CMAKE_FLAGS+=(-DLLVM_USE_HOST_TOOLS=False)

 # Julia expects the produced LLVM tools to be installed into tools and not bin
 # We can't simply move bin to tools since on MingW64 it will also contain the shlib.
